documentation/modules/deploying/proc-deploy-setup-external-clients.adoc

```diff
@@ -0,0 +1,252 @@
+For example, `nodeport` is the least secure, so you might not wish to use it in a production environment.
+
+In this procedure:
+
+. An external listener is configured for the Kafka cluster, with TLS encryption and authentication, and Kafka _simple authorization_ is enabled.
+. A `KafkaUser` is created for the client, with TLS authentication and Access Control Lists (ACLs) defined for _simple authorization_.
+
+You can configure your listener to use TLS or SCRAM-SHA authentication,
+both of which can be used with TLS encryption.
+If you are using an authorization server, you can use token-based link:{BookURLUsing}#assembly-oauth-authentication_str[{oauth} authentication] and link:{BookURLUsing}#assembly-oauth-authorization_str[{oauth} authorization].
+
+When you configure the `KafkaUser` authentication and authorization mechanisms, ensure they match the equivalent Kafka configuration:
+
+* `KafkaUser.spec.authentication` matches `Kafka.spec.kafka.listeners.*.authentication`
+* `KafkaUser.spec.authorization` matches `Kafka.spec.kafka.authorization`
+
+NOTE: Authentication between Kafka users and Kafka brokers depends on the authentication settings for each.
+For example, it is not possible to authenticate a user with TLS if it is not also enabled in the Kafka configuration.
+
+Strimzi operators automate the configuration process:
+
+* The Cluster Operator creates the listeners and sets up the cluster and client certificate authority (CA) certificates to enable authentication within the Kafka cluster.
+* The User Operator creates the user representing the client and sets up the user CA certificates for secure access to the Kafka cluster.
+
+In this procedure, the certificates generated by the Cluster Operator are used, but you can replace them by link:{BookURLUsing}#installing-your-own-ca-certificates-str[installing your own certificates].
+You can also configure your listener to link:{BookURLUsing}#kafka-listener-certificates-str[use a Kafka listener certificate managed by an external Certificate Authority].
+
+Certificates are available in PKCS #12 format (.p12) and PEM (.crt) formats.
+
+.Prerequisites
+
+* The Kafka cluster is available for the client
+* The Cluster Operator and User Operator are running in the cluster
+* A client outside the Kubernetes cluster to connect to the Kafka cluster
+
+.Procedure
+
+. Configure the Kafka cluster with an `external` Kafka listener.
++
+* Define the authentication required to access the Kafka broker through the listener
+* Enable authorization on the Kafka broker
++
+For example:
++
+[source,yaml,subs="+quotes,attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+spec:
+  kafka:
+    # ...
+    listeners: <1>
+      external:
+        type: _LISTENER-TYPE_ <2>
+        tls: true <3>
+        authentication:
+          type: tls <4>
+        configuration:
+          preferredAddressType: InternalDNS <5>
+        networkPolicyPeers: <6>
```

Comment:
not sure about the `networkPolicyPeers` section here. Are we describing how to access the Kafka cluster from outside of Kubernetes right? So it means that the Kafka client is running completely outside of Kubernetes and not in a pod in a different namespace. @scholzj am I missing something?

Reply:
Technically, the rules can cover for example source IP addresses. But I would leave it out as it would not always work and might be confusing.

Reply:
Thanks. I've dropped that property/callout description.